File: domain/Reaccionables/Post.py
from datetime import datetime
from typing import List

from domain.Bandas.Usuario import Usuario
from domain.Reaccionables.Reaccion import Reaccion
from domain.Reaccionables.Reaccionable import Reaccionable


class Post (Reaccionable):
    def __init__(self, reacciones: List[Reaccion], fecha: datetime, titulo: str, contenido: str, autor: Usuario) -> None:
        super().__init__(reacciones)
        self.fecha: datetime = fecha
        self.titulo: str = titulo
        self.contenido: str = contenido
        self.autor: Usuario = autor

File: src/main/java/entities/usuarios/reaccionables/Post.java
package entities.usuarios.reaccionables;

import com.google.gson.annotations.Expose;
import entities.usuarios.Usuario;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import java.util.List;
@Entity
@Table(name = "posts")
public class Post extends Reaccionable {
    @Column @Expose
    private String contenido;
    @ManyToOne @Expose
    private Usuario autor;

    public Post(String contenido, List<Reaccion> reacciones, Usuario autor) {
        super(reacciones);
        this.contenido = contenido;
        this.autor = autor;
    }

    public Post(){}

    public String getContenido() {
        return contenido;
    }

    public void setContenido(String contenido) {
        this.contenido = contenido;
    }

    public Usuario getAutor() {
        return autor;
    }

    public void setAutor(Usuario autor) {
        this.autor = autor;
    }
}
